### PR TITLE
Added top comments describing memories of Combined chain

### PR DESCRIPTION
**Pull request by tomalin**
This is a trivial PR, consisting only of a 2-3 line comment added to the top of each of the memory classes in the Combined Chain, explaining what it is, since it's good practise to have a description of each class.

### CODE REVIEW

File: TrackletAlgorithm/DTCStubMemory.h [hunk 1]
```diff
@@ -4,8 +4,12 @@
 #include "Constants.h"
 #include "MemoryTemplate.h"
 
-// Stub coming from the DTC  
-// does not container header ... i.e. [without valid + layer id]
+// Memory containing stubs written by the InputRouter. 
+// (Unlike the stubs read by the InputRouter, these do not contain the "valid + layer id" bits).
+// N.B. Both class DTCStubMemory & InputStubMemory represent the same BRAM between InputRouter & VMRouter,
+// but DTCStubMemory does not describe the bit fields, so can hold stubs in any of disk/barrel/PS/2S,
+// so is used for the InputRouter output.
```
Comment (aehart) [inline TrackletAlgorithm/DTCStubMemory.h:11]: @tomalin
```suggestion
// (Unlike the stubs read by the InputRouter, these do not contain the
// "valid + layer id" bits).
// N.B. Both class DTCStubMemory & InputStubMemory represent the same BRAM
// between InputRouter & VMRouter, but DTCStubMemory does not describe the bit
// fields, so can hold stubs in any of disk/barrel/PS/2S, so is used for the
// InputRouter output.
```

File: TrackletAlgorithm/FullMatchMemory.h [hunk 2]
```diff
@@ -5,6 +5,9 @@
 #include "MemoryTemplate.h"
 #include "globalFunctions.h"
 
+// Memory written by the MatchProcessor, containing the association between a stub and a tracklet,
+// in a layer that the tracklet seed has been projected to and a matching stub found.
```
Comment (aehart) [inline TrackletAlgorithm/FullMatchMemory.h:9]: @tomalin
```suggestion
// Memory written by the MatchProcessor, containing the association between a
// stub and a tracklet, in a layer that the tracklet seed has been projected to
// and a matching stub found.
```

File: TrackletAlgorithm/InputStubMemory.h [hunk 3]
```diff
@@ -4,6 +4,11 @@
 #include "Constants.h"
 #include "MemoryTemplate.h"
 
+// Memory containing stubs read by the VMRouter.
+// N.B. Both class DTCStubMemory & InputStubMemory represent the same BRAM between InputRouter & VMRouter, 
+// but DTCStubMemory does not describe the bit fields, so can hold stubs in any of disk/barrel/PS/2S,
+// so is used for the InputRouter output.
```
Comment (aehart) [inline TrackletAlgorithm/InputStubMemory.h:10]: @tomalin
```suggestion
// N.B. Both class DTCStubMemory & InputStubMemory represent the same BRAM
// between InputRouter & VMRouter, but DTCStubMemory does not describe the bit
// fields, so can hold stubs in any of disk/barrel/PS/2S, so is used for the
// InputRouter output.
```

File: TrackletAlgorithm/TrackFitMemory.h [hunk 4]
```diff
@@ -4,6 +4,9 @@
 #include "Constants.h"
 #include "MemoryTemplate.h"
 
+// Memory written by the TrackBuilder, containing the track candidates found by the tracklet 
+// pattern reconstruction chain.
```
Comment (aehart) [inline TrackletAlgorithm/TrackFitMemory.h:8]: @tomalin
```suggestion
// Memory written by the TrackBuilder, containing the track candidates found by
// the tracklet pattern reconstruction chain.
```

File: TrackletAlgorithm/TrackletParameterMemory.h [hunk 5]
```diff
@@ -4,7 +4,8 @@
 #include "Constants.h"
 #include "MemoryTemplate.h"
 
-// Data object definition
+// Memory written by the TrackletProcessor, containing the helix parameters of the tracklet seed.
```
Comment (aehart) [inline TrackletAlgorithm/TrackletParameterMemory.h:7]:
```suggestion
// Memory written by the TrackletProcessor, containing the helix parameters of
// the tracklet seed.
```

File: TrackletAlgorithm/TrackletProjectionMemory.h [hunk 6]
```diff
@@ -5,6 +5,9 @@
 #include "MemoryTemplate.h"
 #include "globalFunctions.h"
 
+// Memory written by the TrackletProcessor, containing the intercepts of the tracklet seed projections 
+// to other tracker layers.
```
Comment (aehart) [inline TrackletAlgorithm/TrackletProjectionMemory.h:9]: @tomalin
```suggestion
// Memory written by the TrackletProcessor, containing the intercepts of the
// tracklet seed projections to other tracker layers.
```

**Comment by aehart** [comment]
@tomalin My only comments on your comments is to make the lines shorter so they are easier to read in a terminal. I included suggestions that keep the line width under 80 characters, which seems to be an unofficial standard.

**Comment by tomalin** [comment]
> @tomalin My only comments on your comments is to make the lines shorter so they are easier to read in a terminal. I included suggestions that keep the line width under 80 characters, which seems to be an unofficial standard.

Thanks @aehart . All your suggestions adopted.

**Review by aehart — APPROVED** [review]
Thanks, @tomalin. Looks good.

**Pull request merged**